### PR TITLE
Continue Vehicle Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Implement sub 4AA B0B

### CODE REVIEW

**Comment by janisozaur** [comment]
@duncanspumpkin I submitted minor fix for GCC/Clang, feel free to integrate it to your own commits.

**Comment by duncanspumpkin** [comment]
Tah thanks

File: src/openloco/things/misc.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#include "../objects/objectmgr.h"
+#include "../objects/steam_object.h"
+#include "../map/tilemgr.h"
+#include "thingmgr.h"
+#include "misc.h"
+
+using namespace openloco;
+using namespace openloco::objectmgr;
+
+steam_object * openloco::exahust::object() const
```
Comment (marijnvdwerf) [inline src/openloco/things/misc.cpp]: typo in `exahust`

File: src/openloco/things/misc.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{
+    return objectmgr::get<steam_object>(object_id & 0x7F);
+}
+
+// 0x0044080C
+exahust* openloco::exahust::create(loc16 loc, uint8_t type)
+{
+    if ((uint16_t)loc.x > 12287 || (uint16_t)loc.y > 12287)
+        return nullptr;
+    auto surface = openloco::map::tilemgr::get(loc.x & 0xFFE0, loc.y & 0xFFE0).surface();
```
Comment (marijnvdwerf) [inline src/openloco/things/misc.cpp]: Guessing this is some flooring?

File: src/openloco/map/tile.h [hunk 3]
```diff
@@ -114,14 +116,42 @@ namespace openloco::map
...
         station_element* as_station() const { return as<station_element, element_type::station>(); }
         building_element* as_building() const { return as<building_element, element_type::building>(); }
         industry_element* as_industry() const { return as<industry_element, element_type::industry>(); }
+        unk1_element* as_unk1() const { return as<unk1_element, element_type::unk_1>(); }
     };
     static_assert(sizeof(tile_element) == 8);
 
+    enum class surface_slope : uint8_t
+    {
+        FLAT = 0x00,
```
Comment (marijnvdwerf) [inline src/openloco/map/tile.h]: Should probably be lowercase. Not sure if it should be an enum, could be the same as `window_flags`

File: src/openloco/map/tilemgr.cpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 #include "tilemgr.h"
 #include "../interop/interop.hpp"
+#include <tuple>
```
Comment (marijnvdwerf) [inline src/openloco/map/tilemgr.cpp]: Does it needs this import? Already done by header.

File: src/openloco/objects/objectmgr.h [hunk 5]
```diff
@@ -101,6 +101,8 @@ namespace openloco::objectmgr
     template<>
     interface_skin_object* get();
     template<>
+    steam_object* get();
```
Comment (marijnvdwerf) [inline src/openloco/objects/objectmgr.h]: Implementation uses different signature: `steam_object* get(size_t id)`

File: src/openloco/map/tile.h (hunk 3 above)
Comment (duncanspumpkin) [reply src/openloco/map/tile.h]: yeah i have to switch this to a constexpr

**Pull request merged**